File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/messaging/TypeConverter.java
package io.lsdconsulting.interceptors.messaging;

import com.fasterxml.jackson.core.JsonProcessingException;

import static lsd.format.json.ObjectMapperCreatorKt.createObjectMapper;

public class TypeConverter {
    static String convertToString(Object payload) {
        String payloadString;
        if (payload instanceof String) {
            payloadString = (String) payload;
        } else if (payload instanceof byte[]) {
            payloadString = new String((byte[]) payload);
        } else {
            try {
                payloadString = createObjectMapper().writeValueAsString(payload);
            } catch (JsonProcessingException e) {
                throw new RuntimeException(e);
            }
        }
        return payloadString;
    }
}

File: lsd-interceptors/src/main/kotlin/io/lsdconsulting/interceptors/messaging/TypeConverter.kt
package io.lsdconsulting.interceptors.messaging

import com.fasterxml.jackson.core.JsonProcessingException
import lsd.format.json.createObjectMapper

// TODO Replace with lsd-formatting-library
fun convertToString(payload: Any?): String =
    when (payload) {
        is String -> {
            payload
        }

        is ByteArray -> {
            String((payload as ByteArray?)!!)
        }

        else -> {
            try {
                createObjectMapper().writeValueAsString(payload)
            } catch (e: JsonProcessingException) {
                throw RuntimeException(e)
            }
        }
    }
